sensorprocess/sensorprocess.go

```diff
@@ -37,6 +38,10 @@ func Start(
 			return false
 		default:
 			if jobDone := addLidarReading(ctx, config); jobDone {
+				err := config.RunFinalOptimizationFunc(ctx, config.Timeout)
```

Comment:
is there anything we need to add for other sensors? like if the lidar completes its own offline dataset before the imu, do we need to block the imu from trying to add sensors? @jeremyrhyde

Reply:
I will add the additional blocking mechanisms i nthe IMU PR. Should be a simple:

```
if jobDone := addLidarReading(ctx, config); jobDone {
        stopped = true
        err := config.RunFinalOptimizationFunc(ctx, config.Timeout)
```